ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveTypeSystemImpl.java

```diff
@@ -37,6 +39,14 @@
   private static final int MAX_BINARY_PRECISION      = Integer.MAX_VALUE;
   private static final int MAX_TIMESTAMP_PRECISION   = 9;
   private static final int MAX_TIMESTAMP_WITH_LOCAL_TIME_ZONE_PRECISION = 15; // Up to nanos
+  private static final int DEFAULT_BOOLEAN_PRECISION  = 1;
```

Comment:
I think you are right. Implementation in Calcite has it defined for Boolean since I think they consider precision as length... However, I do not think it makes much sense. I have removed it.